### DISCUSSION TITLE
How to change KafkaUser certificate validity?

### DISCUSSION THREAD

**Discussion by vanhoale** [strimzi/strimzi-kafka-operator#8155] [Q&A]
I have a Kafka cluster that is using an ingress external listener. I'm using my own CA that has validity for 10 years. A super KafkaUser was created and I see its cert/key/p12 was generated as well, when I check the certificate validity it just has 1 year. I wonder if I can have a configuration option somewhere to have it longer or shorter?

**Comment by scholzj** [comment]
You should be able to configure it in `.spec.clientsCa.validityDays` in the Kafka custom resource: https://strimzi.io/docs/operators/latest/full/configuring.html#con-certificate-renewal-str

**Reply by vanhoale** [reply]
Thanks for the quick response! Does it apply to all KafkaUsers?

**Reply by scholzj** [reply]
Right ... it will apply to all users. But it is used only in the next renewal. You can _force_ the renewal by deleting the certificate secret of the users.

**Reply by vanhoale** [reply]
Awesome, thanks!

**Reply by vanhoale** [reply]
What if we move the grant validity to KafkaUser spec? I mean it can be a new feature

**Reply by scholzj** [reply]
That is currently not supported.

**Reply by vanhoale** [reply]
ok, thanks!
